Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ContentSubgraph.php

```diff
@@ -497,32 +497,109 @@ private function buildChildNodesQuery(NodeAggregateId $parentNodeAggregateId, Fi
         return $queryBuilder;
     }
 
-    private function buildReferencesQuery(bool $backReferences, NodeAggregateId $nodeAggregateId, FindReferencesFilter|FindBackReferencesFilter|CountReferencesFilter|CountBackReferencesFilter $filter): QueryBuilder
+    private function buildReferencesQuery(NodeAggregateId $nodeAggregateId, FindReferencesFilter|CountReferencesFilter $filter): QueryBuilder
     {
-        $sourceTablePrefix = $backReferences ? 'd' : 's';
-        $destinationTablePrefix = $backReferences ? 's' : 'd';
+        $subselectParameters = [
+            'nodeAggregateId' => $nodeAggregateId->value,
+            'contentStreamId' => $this->contentStreamId->value,
+            'dimensionSpacePointHash' => $this->dimensionSpacePoint->hash,
+        ];
+        $subtreeTagConstraints = '';
+        $i = 0;
+        foreach ($this->visibilityConstraints->excludedSubtreeTags as $excludedTag) {
```

Comment:
isnt there the method `addSubtreeTagConstraints` which we still use below?

but for a proper review i would need to dig into this deeply

Reply:
yep; needed to copy it though, since the query builder has no support (that I know of) for subselects.